executor/executor.go

```diff
@@ -1371,41 +1372,65 @@ func (e *MaxOneRowExec) Open(ctx context.Context) error {
 	if err := e.baseExecutor.Open(ctx); err != nil {
 		return err
 	}
+	e.rowData = make([][]byte, 0)
 	e.evaluated = false
 	return nil
 }
 
 // Next implements the Executor Next interface.
 func (e *MaxOneRowExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	req.Reset()
-	if e.evaluated {
+
+	if e.evaluated && len(e.rowData) == 0 {
```

Comment:
Why we need check `len(e.rowData) == 0`? 
I think the logic after this line should only run once..